metpy/calc/thermo.py

```diff
@@ -1461,15 +1464,22 @@ def _find_append_zero_crossings(x, y):
 
     """
     # Find and append crossings to the data
-    crossings = find_intersections(x[1:], y[1:], np.zeros_like(y[1:]) * y.units)
-    x = concatenate((x, crossings[0]))
+    if hasattr(x, 'units'):
+        x_units = x.units
+        x = x.m
+    else:
+        x_units = 1
+    crossings = find_intersections(np.log(x[1:]), y[1:], np.zeros_like(y[1:]) * y.units)
```

Comment:
Can't you use the `log_x` option to `find_intersections` here to simplify things?